src/repr/src/adt/mz_acl_item.rs

```diff
@@ -0,0 +1,347 @@
+                grantor: RoleId::from_proto(grantor)?,
+                acl_mode: AclMode::from_proto(acl_mode)?,
+            }),
+            (None, _, _) => Err(TryFromProtoError::missing_field("ProtoMzAclItem::grantee")),
+            (_, None, _) => Err(TryFromProtoError::missing_field("ProtoMzAclItem::grantor")),
+            (_, _, None) => Err(TryFromProtoError::missing_field("ProtoMzAclItem::acl_mode")),
+        }
+    }
+}
+
+impl Columnation for MzAclItem {
+    type InnerRegion = CloneRegion<MzAclItem>;
+}
+
+#[test]
+fn test_mz_acl_parsing() {
+    let s = "u42=rw/s666";
+    let mz_acl: MzAclItem = s.parse().unwrap();
+    assert_eq!(RoleId::User(42), mz_acl.grantee);
+    assert_eq!(RoleId::System(666), mz_acl.grantor);
+    assert!(!mz_acl.acl_mode.contains(AclMode::INSERT));
+    assert!(mz_acl.acl_mode.contains(AclMode::SELECT));
+    assert!(mz_acl.acl_mode.contains(AclMode::UPDATE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::DELETE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::USAGE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::CREATE));
+    assert_eq!(s, mz_acl.to_string());
+
+    let s = "=UC/u4";
+    let mz_acl: MzAclItem = s.parse().unwrap();
+    assert_eq!(RoleId::Public, mz_acl.grantee);
+    assert_eq!(RoleId::User(4), mz_acl.grantor);
+    assert!(!mz_acl.acl_mode.contains(AclMode::INSERT));
+    assert!(!mz_acl.acl_mode.contains(AclMode::SELECT));
+    assert!(!mz_acl.acl_mode.contains(AclMode::UPDATE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::DELETE));
+    assert!(mz_acl.acl_mode.contains(AclMode::USAGE));
+    assert!(mz_acl.acl_mode.contains(AclMode::CREATE));
+    assert_eq!(s, mz_acl.to_string());
+
+    let s = "s7=/s12";
+    let mz_acl: MzAclItem = s.parse().unwrap();
+    assert_eq!(RoleId::System(7), mz_acl.grantee);
+    assert_eq!(RoleId::System(12), mz_acl.grantor);
+    assert!(!mz_acl.acl_mode.contains(AclMode::INSERT));
+    assert!(!mz_acl.acl_mode.contains(AclMode::SELECT));
+    assert!(!mz_acl.acl_mode.contains(AclMode::UPDATE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::DELETE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::USAGE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::CREATE));
+    assert_eq!(s, mz_acl.to_string());
+
+    let s = "=/u100";
+    let mz_acl: MzAclItem = s.parse().unwrap();
+    assert_eq!(RoleId::Public, mz_acl.grantee);
+    assert_eq!(RoleId::User(100), mz_acl.grantor);
+    assert!(!mz_acl.acl_mode.contains(AclMode::INSERT));
+    assert!(!mz_acl.acl_mode.contains(AclMode::SELECT));
+    assert!(!mz_acl.acl_mode.contains(AclMode::UPDATE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::DELETE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::USAGE));
+    assert!(!mz_acl.acl_mode.contains(AclMode::CREATE));
+    assert_eq!(s, mz_acl.to_string());
+
+    assert!("u32=C/".parse::<MzAclItem>().is_err());
+    assert!("=/".parse::<MzAclItem>().is_err());
+    assert!("f62hfiuew827fhh".parse::<MzAclItem>().is_err());
```

Comment:
Add something with multiple `=` and `/` too.

Reply:
Done